components/brave_wallet/common/serializer_stream.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_COMMON_SERIALIZER_STREAM_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_COMMON_SERIALIZER_STREAM_H_
+
+#include <vector>
+
+#include "base/containers/span.h"
+#include "base/memory/raw_ref.h"
+
+namespace brave_wallet {
+
+class SerializerStream {
+ public:
+  explicit SerializerStream(std::vector<uint8_t>* to) : to_(to) {}
```

Comment:
Let's name it `BtcLikeSerializerStream` or `BitcoinLikeSerializerStream` with some comment that this only for BTC and ZEC